### PR TITLE
Add minimath support

### PR DESCRIPTION
**Pull request by johnflux**

### CODE REVIEW

File: model/model_training/custom_datasets/instruction.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 """
     These are in the form of 'INSTRUCTION', 'RESPONSE'
 """
-from custom_datasets.formatting import format_pair
+from custom_datasets.formatting import format_pairs
```
Comment (sanagno) [inline model/model_training/custom_datasets/instruction.py:4]: remove this for new formatting

File: model/model_training/custom_datasets/instruction.py [hunk 2]
```diff
@@ -17,20 +17,21 @@
...
+        self.response_column = "RESPONSE" if dataset != "minimath" else "answer"
         self.max_words = max_words
 
     def __len__(self):
         return len(self.dataset)
 
     def __getitem__(self, idx):
         data = self.dataset[idx]
-        return format_pair((data[self.instruction_column], data[self.response_column]))
+        return format_pairs((data[self.instruction_column], data[self.response_column]))
```
Comment (sanagno) [inline model/model_training/custom_datasets/instruction.py:37]: replace with 
`return (data[self.instruction_column], data[self.response_column])`

File: model/model_training/custom_datasets/instruction.py (hunk 1 above)
Comment (johnflux) [reply model/model_training/custom_datasets/instruction.py]: Done

**Comment by sanagno** [comment]
https://huggingface.co/datasets/kentsui/minimath: are the data encoded correctly as strings?

**Comment by johnflux** [comment]
> https://huggingface.co/datasets/kentsui/minimath: are the data encoded correctly as strings?

No, they are encoded as bytes :-/

File: model/model_training/models/patching.py [hunk 3]
```diff
@@ -79,7 +80,7 @@ def add_flash_attn(module: nn.Module, causal: bool = True):
     Limitations:
       - Only works for fp16 or bf16 inputs
       - Requires inputs to be on CUDA
-      - `outptu_attentions=True` does work after patching, attention weights will be None
+      - `output_attentions=True` does work after patching, attention weights will be None
```
Comment (dvruette) [inline model/model_training/models/patching.py:83]: typo from my side: 
```suggestion
      - `output_attentions=True` does not work after patching, attention weights will be None
```

**Review by dvruette — APPROVED** [review]
LGTM, can be merged with code owner approval

**Comment by sanagno** [comment]
Are the results you got with this dataset in bytes format?

**Comment by johnflux** [comment]
Yes

**Pull request merged**